thoth/sourcemanagement/sourcemanagement.py

```diff
@@ -58,11 +58,15 @@ def __init__(
 
         if self.service_type == ServiceType.GITHUB:
             self.service = GithubService(self.token)
+            if service_url != None:
+                self.service = GithubService(self.token, instance_url=service_url)
```

Comment:
It might be more efficient to create one object based on parameters supplied, rather than creating one and immediately discarding it with another one:

```python
            if service_url != None:
                self.service = GithubService(self.token, instance_url=service_url)
            else:
                self.service = GithubService(self.token)
```